cmd/kubeadm/app/cmd/upgrade/common.go

```diff
@@ -181,6 +186,19 @@ func runPreflightChecks(ignorePreflightErrors sets.String) error {
 	return preflight.RunRootCheckOnly(ignorePreflightErrors)
 }
 
+// runCoreDNSpreflightChecks runs the CoreDNS migration preflight checks.
+func runCoreDNSpreflightChecks(client clientset.Interface, ignorePreflightErrors sets.String) error {
+	currentDNSType, _, err := dns.DeployedDNSAddon(client)
+	if err != nil {
+		return err
+	}
+	if currentDNSType == kubeadmapi.CoreDNS {
+		fmt.Println("[preflight] Running CoreDNS migration pre-flight checks.")
+		return upgrade.RunCoreDNSMigrationCheck(client, ignorePreflightErrors)
```

Comment:
```
currentDNSType, _, err := dns.DeployedDNSAddon(client)
	if err != nil {
		return err
	}
	if currentDNSType == kubeadmapi.CoreDNS {
...
```

can be inside `upgrade.RunCoreDNSMigrationCheck`, and optionally make the function a no-op if kube-dns is used.
(i.e. exit early and don't run `preflight.RunChecks(migrationChecks, os.Stderr, ignorePreflightErrors)`)